### PR TITLE
LCH-4486-1.x: Ability to take snapshots from plexus

### PR DESCRIPTION
**Pull request by ankitsingh0188**

### CODE REVIEW

File: src/ContentHub.php [hunk 1]
```diff
@@ -583,4 +583,62 @@ protected function getResponseJson(RequestInterface $request)
...
+     * Fetch snapshots.
+     *
+     * @return mixed
+     *   Response.
+     *
+     * @throws \Exception
+     */
+    public function getSnapshots()
+    {
+      return self::getResponseJson($this->get('snapshots'));
```
Comment (Ateee89) [inline src/ContentHub.php]: I am not sure how can this work. getResponseJson() function first parameter should be a RequestInterface. $this->get('snapshots') will return a ResponseInterface

File: src/ContentHub.php [hunk 2]
```diff
@@ -583,4 +583,62 @@ protected function getResponseJson(RequestInterface $request)
...
+     * Create a snapshot.
+     *
+     * @return mixed
+     *   Response.
+     *
+     * @throws \Exception
+     */
+    public function createSnapshot()
+    {
+      return self::getResponseJson($this->post('snapshots'));
```
Comment (Ateee89) [inline src/ContentHub.php]: Same here

File: src/ContentHub.php [hunk 3]
```diff
@@ -583,4 +583,62 @@ protected function getResponseJson(RequestInterface $request)
...
+     *   The name of the snapshot.
+     *
+     * @return \Psr\Http\Message\ResponseInterface
+     *   Response.
+     *
+     * @throws \GuzzleHttp\Exception\RequestException
+     */
+    public function deleteSnapshot($name)
+    {
+      return self::getResponseJson($this->delete("snapshots/$name"));
```
Comment (Ateee89) [inline src/ContentHub.php]: Same here

File: src/ContentHub.php [hunk 4]
```diff
@@ -583,4 +583,62 @@ protected function getResponseJson(RequestInterface $request)
...
+     *   The name of the snapshot.
+     *
+     * @return \Psr\Http\Message\ResponseInterface
+     *   Response.
+     *
+     * @throws \GuzzleHttp\Exception\RequestException
+     */
+    public function restoreSnapshot($name)
+    {
+      return self::getResponseJson($this->put("snapshots/$name/restore"));
```
Comment (Ateee89) [inline src/ContentHub.php]: Same here.

**Review by Ateee89 — CHANGES_REQUESTED** [review]

File: src/ContentHub.php [hunk 5]
```diff
@@ -583,4 +583,70 @@ protected function getResponseJson(RequestInterface $request)
...
+     * Fetch snapshots.
+     *
+     * @return mixed
+     *   Response.
+     *
+     * @throws \Exception
+     */
+    public function getSnapshots()
+    {
+      $endpoint = "$this->api_version/snapshots";
```
Comment (abarriosr) [inline src/ContentHub.php]: The only available version for this API call is `/v2`. Then all calls should use that instead of `$api_version`.

**Review by Ateee89 — APPROVED** [review]

**Review by Ateee89 — APPROVED** [review]

**Pull request merged**